File: app/src/main/java/com/platdmit/simplecloudmanager/domain/converters/BackupConverter.java
package com.platdmit.simplecloudmanager.domain.converters;

import com.platdmit.simplecloudmanager.data.api.models.ApiBackup;
import com.platdmit.simplecloudmanager.data.database.entity.DbBackup;
import com.platdmit.simplecloudmanager.domain.models.Backup;

import java.util.List;

public interface BackupConverter {
    DbBackup fromApiToDb(ApiBackup apiBackup, long serverId);
    Backup fromDbToDomain(DbBackup dbBackup);
    List<Backup> fromDbToDomainList(List<DbBackup> dbList);
    List<DbBackup> fromApiToDbList(List<ApiBackup> apiList, long serverId);
}

File: app/src/main/java/com/platdmit/simplecloudmanager/domain/converters/BackupConverter.kt
package com.platdmit.simplecloudmanager.domain.converters

import com.platdmit.data.api.models.ApiBackup
import com.platdmit.data.database.entity.DbBackup
import com.platdmit.simplecloudmanager.domain.models.Backup

interface BackupConverter {
    fun fromApiToDb(apiBackup: ApiBackup, serverId: Long): DbBackup
    fun fromDbToDomain(dbBackup: DbBackup): Backup
    fun fromDbToDomainList(dbList: List<DbBackup>): List<Backup>
    fun fromApiToDbList(apiList: List<ApiBackup>, serverId: Long): List<DbBackup>
}